ArjunaJTS/integration/src/test/java/com/arjuna/ats/jta/distributed/spi/SPIUnitTest.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.arjuna.ats.jta.distributed.spi;
+
+import com.arjuna.ats.arjuna.common.RecoveryEnvironmentBean;
+
+import com.arjuna.ats.internal.jts.ORBManager;
+import com.arjuna.ats.jta.common.JTAEnvironmentBean;
+import com.arjuna.ats.jta.common.jtaPropertyManager;
+import com.arjuna.ats.jta.distributed.JndiProvider;
+import com.arjuna.ats.jta.utils.JNDIManager;
+import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
+import com.arjuna.orbportability.common.OrbPortabilityEnvironmentBean;
+import com.arjuna.orbportability.internal.utils.PostInitLoader;
+import org.jboss.tm.TransactionManagerLocator;
+import org.jboss.tm.listener.EventType;
+import org.jboss.tm.listener.TransactionListenerRegistry;
+import org.jboss.tm.listener.TransactionListenerRegistryLocator;
+import org.jboss.tm.usertx.UserTransactionRegistry;
+import org.jboss.tm.usertx.client.ServerVMClientUserTransaction;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.naming.InitialContext;
+import javax.transaction.*;
+import java.util.*;
+import com.arjuna.orbportability.OA;
+import com.arjuna.orbportability.ORB;
+import com.arjuna.orbportability.RootOA;
+import org.omg.CORBA.ORBPackage.InvalidName;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertTrue;
+
+public class SPIUnitTest {
+    private static InitialContext initialContext;
+    private static ORB myORB = null;
+    private static RootOA myOA = null;
+
+    private static void initORB() throws InvalidName {
+        OrbPortabilityEnvironmentBean opeb =  BeanPopulator.getDefaultInstance(OrbPortabilityEnvironmentBean.class);
+
+        // TODO
```

Comment:
I guess so - I need to make sure the orb is initialized for all the orbs that we test against.